Task 2/Task 2.1/CustomString/Class1.cs

```diff
@@ -0,0 +1,187 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace MyString
+{
+    public class CustomString : IEquatable<CustomString>
+    {
+        public CustomString(string text)
+        {
+            MyString = text.ToCharArray();
+        }
+
+        private char[] _myString;
+
+        public char[] MyString
+        {
+            get
+            {
+                return _myString;
+            }
+            set
+            {
+                _myString = value;
+            }
+        }
+
+        public static CustomString operator + (CustomString str1, CustomString str2)
+        {
+            string result = String.Empty;
+
+            foreach (char item in str1.MyString)
+            {
+                result += item;
+            }
+
+            foreach (char item in str2.MyString)
+            {
+                result += item;
+            }
+
+            return new CustomString(result);
+        }
+
+        public static bool operator == (CustomString str1, CustomString str2)
+        {
+            bool flag = true;
+            if (str1.MyString.Length != str2.MyString.Length)
+            {
+                flag = false;
+                return flag;
+            }
+            else
+            {
+                for (int i = 0; i < str1.MyString.Length; i++)
+                {
+                    if (!Equals(str1.MyString[i], str2.MyString[i])) flag = false;
+                }
+
+                return flag;
+            }
+        }
+
+        public static bool operator != (CustomString str1, CustomString str2)
+        {
+            bool flag = false;
+            if (str1.MyString.Length != str2.MyString.Length)
+            {
+                flag = true;
+                return flag;
+            }
+            else
+            {
+                for (int i = 0; i < str1.MyString.Length; i++)
+                {
+                    if (!Equals(str1.MyString[i], str2.MyString[i])) 
+                    {
+                        flag = true;
+                    }
+                }
+
+                return flag;
+            }
+        }
+
+        /// <summary>
+        /// Метод, определяющий, есть ли искомый символ в кастомной строке.
+        /// </summary>
+        /// <param name="Symbol"></param>
+        /// <returns>Возвращает true или false, в зависимости от наличия символа в строке.</returns>
+        public bool SearchSymbol(char Symbol)
+        {
+            bool flag = false;
+            foreach (char item in MyString)
+            {
+                if (item == Symbol)
+                    flag = true;
+            }
+            return flag;
+        }
```

Comment:
1. When method is called "Find" or "Search" you usually expect that the method will return info about where something is. (an index for example)\
In this case the method should be named `ContainsSymbol`
2. parameter names start with lowerCase
3. `flag` variable is redundant, you cant return fast right in `foreach` loop